File: osgi-framework-bootstrap/src/test/java/net/corda/osgi/framework/OSGiFrameworkFactoryMock.java
package net.corda.osgi.framework;

import org.osgi.framework.launch.Framework;
import org.osgi.framework.launch.FrameworkFactory;
import java.util.Map;

public final class OSGiFrameworkFactoryMock implements FrameworkFactory {

    @Override
    public Framework newFramework(Map<String, String> configurationMap) {
        return new OSGiFrameworkMock(configurationMap);
    }

}

File: osgi-framework-bootstrap/src/test/kotlin/net/corda/osgi/framework/OSGiFrameworkFactoryMock.kt
package net.corda.osgi.framework

import org.osgi.framework.launch.Framework
import org.osgi.framework.launch.FrameworkFactory

class OSGiFrameworkFactoryMock: FrameworkFactory {

    override fun newFramework(configurationMap: MutableMap<String, String>): Framework {
        return OSGiFrameworkMock(configurationMap)
    }

}